numpyro/distributions/continuous.py

```diff
@@ -715,35 +715,44 @@ def variance(self):
         return np.broadcast_to(var, self.batch_shape)
 
 
-@copy_docs_from(Distribution)
-class TruncatedCauchy(Distribution):
-    arg_constraints = {'low': constraints.real, 'loc': constraints.real,
-                       'scale': constraints.positive}
-    reparametrized_params = ['low', 'loc', 'scale']
+class _BaseTruncatedCauchy(Distribution):
+    # NB: this is a truncated cauchy with low=0, scale=1
+    support = constraints.positive
 
-    def __init__(self, low=0., loc=0., scale=1., validate_args=None):
-        self.low, self.loc, self.scale = promote_shapes(low, loc, scale)
-        batch_shape = lax.broadcast_shapes(np.shape(low), np.shape(loc), np.shape(scale))
-        super(TruncatedCauchy, self).__init__(batch_shape=batch_shape, validate_args=validate_args)
+    def __init__(self, base_loc):
+        self.base_loc = base_loc
+        super(_BaseTruncatedCauchy, self).__init__(batch_shape=np.shape(base_loc))
 
     def sample(self, key, sample_shape=()):
         # We use inverse transform method:
         # z ~ inv_cdf(U), where U ~ Uniform(cdf(low), cdf(high)).
         #                         ~ Uniform(arctan(low), arctan(high)) / pi + 1/2
         size = sample_shape + self.batch_shape
-        low = (self.low - self.loc) / self.scale
-        minval = np.arctan(low)
+        minval = -np.arctan(self.base_loc)
         maxval = np.pi / 2
         u = minval + random.uniform(key, shape=size) * (maxval - minval)
-        return self.loc + np.tan(u) * self.scale
+        return self.base_loc + np.tan(u)
 
     def log_prob(self, value):
         if self._validate_args:
             self._validate_sample(value)
-        low = (self.low - self.loc) / self.scale
         # pi / 2 is arctan of self.high when that arg is supported
-        normalize_term = np.log(np.pi / 2 - np.arctan(low)) + np.log(self.scale)
-        return - np.log1p(((value - self.loc) / self.scale) ** 2) - normalize_term
+        normalize_term = np.log(np.pi / 2 + np.arctan(self.base_loc))
+        return - np.log1p((value - self.base_loc) ** 2) - normalize_term
```

Comment:
@neerajprad to verify the equivalent of these formulas, we only need to replace low -> 0, scale -> 1, loc -> base_loc. Anyway, the current tests already covert it.

Reply:
Thanks, that makes sense.